processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -435,487 +419,58 @@ public Cursor apply(final Interval inputInterval)
                                         );
 
 
-                  final Offset initOffset = offset.clone();
+                  final Offset cursorOffset = offset.clone();
```

Comment:
I simplified this area a bit and removed anonymous subclasses. Also renamed the variable to `baseCursorOffset`. "initOffset" is not quite right because in the previous version of the code, "initOffset" was immutable and always pointing to the "init offset" (and cloned). Now this "cursorOffset" is actually mutable and resettable inside the cursor.